sg/deps.go

```diff
@@ -22,6 +22,13 @@ func Deps(ctx context.Context, functions ...interface{}) {
 	var wg sync.WaitGroup
 	for i, f := range checkedFunctions {
 		i, f := i, f
+		dependencies := getDependencies(ctx)
+		for _, dependency := range dependencies {
+			if dependency.ID() == f.ID() {
+				panic("dependency cycle!")
```

Comment:
Will it print what dependency causes the cycle here? Not sure how panic works with stack traces for go

Reply:
It might be deductible from the stack trace, but it would make sense to include the names of dependencies in the chain in the error string, as well as the wanted function. I'll add that 👍

Reply:
I realized that I don't have the entry function in the dependency list, so the output could be a bit weird. But at least it prints a hint of where the dependency cycle is 😅 See updated commit message and PR description.